README.md

```diff
@@ -62,13 +66,16 @@ IPC Diagnose: Beckhoff IPC’s provide a web interface for diagnose and configur
 - (optional) enable bhf::ads::SetLocalAddress() and set to the AMS NetId you choose for the ADS client (e.g. 192.168.0.1.1.1).
 
 ## configure meson to build example into "build" dir
+```
 meson example/build example
-
+```
 ## let ninja build the example
+```
 ninja -C example/build
-
+```
 ## and run the example
+```
 ./example/build/example
-
+```
```

Comment:
There is "CommonMark" as standardization which is supported by Github/Gitlab/StackOverflow/... so using the basics of markdown will not give any conflict, yet making it more pleasant to work with.

Atomic commits makes reviewing sometimes more easy, to see stepwise what has happened. If you don't want to overflow your history, you can always ["squash and merge" on pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges). But off course this can also be done by the contributor if this is preferred. 

It might also be a good option to put all your preference for contributions in a CONTRIBUTION.md file so people that are willing to put time and effort in contribution can check some project guidelines first before doing the work in a way that the maintainer doesn't like. [Basics of CONTRIBUTION.md](https://mozillascience.github.io/working-open-workshop/contributing/). Since it is an open source project, making use of open source concepts will probably be beneficial for the project.

I do get your point though and the choises are up to you

Reply:
Well, even in our tiny README, we already use features not defined in CommonMark (shell annotation). And its sad to read  six guys started in 2014 and not even a decade later half of them left.

However, I rebased your stuff and put another cleanup on top see #192. If it is okay for you I will merge the heavily changed commit under your name https://github.com/Beckhoff/ADS/pull/192/commits/460472d3ad8ce6e3fab35b363910828197a8d4f2

Yeah, you are right about the "squash and merge". I could have used this in this PR (if I didn't had to rebase).  But I am used to a rebase PR regulary as GitLab automatically tracks the changes for you, so you can see the diffs between revisions. And in general I don't want to touch commits from other people, see when I squash your commits, I have to rewrite the commit message and basically change the thing you signed-off on.

CONTRIBUTING.md, absolutely it is on my TODO list. But it is very hard to reach the bottom where I put it... I moved it a few lines to the top ;-).

Reply:
Looks good to me.

Whoops, somehow overlooked the CONTRIBUTING file, my apologies!

Valid point on the shell annotation as well. I guess this is, as you mentioned, a subset I gotten used to. (it makes the block directly runnable with a single click on most IDE's) Thanks for all the feedback. ;)